app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java

```diff
@@ -481,6 +505,48 @@ private boolean hasSkippedAccount() {
         return sp.getBoolean(MainActivity.PREFERENCE_KEY_INTERNAL_SKIP_WALLET_ACCOUNT, false);
     }
 
+    public void launchMoonpayFlow() {
+        Context context = getContext();
+        String receiveAddress = null;
+        if (context != null) {
+            try {
+                SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(context);
+                receiveAddress = sp.getString(MainActivity.PREFERENCE_KEY_INTERNAL_WALLET_RECEIVE_ADDRESS, null);
+                if (Helper.isNullOrEmpty(receiveAddress)) {
+                    showError(getString(R.string.receive_address_not_set));
+                    return;
+                }
+
+                long userId = Lbryio.currentUser != null ? Lbryio.currentUser.getId() : 0;
```

Comment:
what happens if the user_id is 0 when passed to moonpay?

Reply:
Nothing bad happens. The `externalCustomerId` field is freeform and it's just a way we can use tell that a purchase was done on Android.